libethcore/EthashGPUMiner.cpp

```diff
@@ -85,14 +91,33 @@ class EthashCLHook: public ethash_cl_miner::search_hook
 //		std::cerr << "Searched " << _count << " from " << _startNonce << std::endl;
 		m_owner->accumulateHashes(_count);
 		m_last = _startNonce + _count;
-		if (m_abort || m_owner->shouldStop())
-			return (m_aborted = true);
-		return false;
+		//if (m_abort || m_owner->shouldStop())
+		//	return (m_aborted = true);
+		//return false;
+		// TODO: still need a way to stop the GPU thread
```

Comment:
The change's that i was doing up doesn't quite fit in perfectly with your changes, i am changing the way the kernels are run from the current way. Instead of running 1 kernel then running 2nd kernel before reading the results of 1st, then swap/swap... changing it over to use one main kernel, and one that halts/aborts the job.

So rather than having the kernel finishing and coming back to CPU so often, when a job comes in that miner is created and starts working, it won't ever stop working until you abort it's execution - the idea being that on the CPU you are notified of newWork and simply fire off the abort kernel to kill off the current running one, then queue up this job and fire it off.

How i'm accomplishing this with the kernels:
1) Create two command queues in a context.
2) Create two kernels, one to do the work and another to halt execution. Each kernel has access to a shared global buffer.
3) Load the first kernel into queue1.
4) Load the second kernel into queue2 when we want to halt execution.

(Alternatively we could use an out-of-order queue and load the second kernel into the same command queue to halt execution -- But i think that would be more trouble because we have to then be very careful and use clFinish/clFlush as required/necessary. However it may be a more natural way of writing/doing this... what do you think?